File: src/main/java/eterea/api/rest/model/ClienteMovimientoPrevio.java
package eterea.api.rest.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;
import org.hibernate.Hibernate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.Objects;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity
@Table(name = "clientemovimientoprevio")
@AllArgsConstructor
public class ClienteMovimientoPrevio extends Auditable implements Serializable {

	@Serial
	private static final long serialVersionUID = -4005223853384447248L;

	@Id
	@Column(name = "clientemovimientoprevio_id")
	private Long clienteMovimientoPrevioId;
	
	@Column(name = "en_contruccion")
	private Byte enConstruccion;
	
	@Column(name = "tiene_articulos")
	private Byte articulos;
	
	@Column(name = "ticket_impreso")
	private Byte ticketImpreso;
	
	@Column(name = "cliente_id")
	private Long clienteId;
	
	@Column(name = "punto_venta")
	private Integer puntoVenta;
	
	@Column(name = "fecha")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fecha;
	
	@Column(name = "importe")
	private BigDecimal importe = BigDecimal.ZERO;
	
	@Column(name = "clientemovimiento_id")
	private Long clienteMovimientoId;

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
		ClienteMovimientoPrevio that = (ClienteMovimientoPrevio) o;
		return clienteMovimientoPrevioId != null && Objects.equals(clienteMovimientoPrevioId, that.clienteMovimientoPrevioId);
	}

	@Override
	public int hashCode() {
		return getClass().hashCode();
	}
}

File: src/main/java/eterea/core/service/kotlin/model/ClienteMovimientoPrevio.kt
package eterea.core.service.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.json.JsonMapper
import jakarta.persistence.*
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(name = "clientemovimientoprevio")
data class ClienteMovimientoPrevio(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "clientemovimientoprevio_id")
    var clienteMovimientoPrevioId: Long? = null,

    var enConstruccion: Byte = 0,
    var tieneArticulos: Byte = 0,
    var ticketImpreso: Byte = 0,
    var clienteId: Long? = null,
    var puntoVenta: Int? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fecha: OffsetDateTime? = null,

    var importe: BigDecimal = BigDecimal.ZERO,

    @Column(name = "clientemovimiento_id")
    var clienteMovimientoId: Long? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "clienteId", insertable = false, updatable = false)
    var cliente: Cliente? = null,

    @OneToMany(fetch = FetchType.EAGER)
    @JoinColumn(name = "clientemovimientoprevio_id", insertable = false, updatable = false)
    var articuloMovimientoPrevios: List<ArticuloMovimientoPrevio>? = null

) : Auditable() {

    fun jsonify(): String {
        try {
            return JsonMapper
                .builder()
                .findAndAddModules()
                .build()
                .writerWithDefaultPrettyPrinter()
                .writeValueAsString(this)
        } catch (e: JsonProcessingException) {
            return "jsonify error -> ${e.message}"
        }
    }

}
